### PR TITLE
Gradle: Improve configuration & TestClient build

### PR DESCRIPTION
**Pull request by xerus2000**
Even if you don't understand it, please have a glance at it and run a few tasks locally
- `:test-client:jar`, if run independently, should fail on master but work on this branch
- check the deployed targets

You probably won't notice any big differences, but architecturally the code is now a lot cleaner and my gradle understanding improved a lot as well after hours of tinkering with it. Feel free to ask what the changes do, too.

### CODE REVIEW

File: helpers/test-client/build.gradle.kts [hunk 1]
```diff
@@ -10,11 +10,11 @@ sourceSets.main {
 }
 
 application {
-    mainClassName = "sc.TestClient"
+    mainClass.set("sc.TestClient")
 }
 
 dependencies {
-    implementation(project(":plugin"))
+    // TODO this dependency is only for accessing the Configuration, remove it
```
Comment (SKoschnicke) [inline helpers/test-client/build.gradle.kts:17]: remove it... now? Or when?

**Review by SKoschnicke — COMMENTED** [review]
`:test-client:jar` does not fail in master.

In master (`17ca22ab`):
```
$ ./gradlew :test-client:jar
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=on -Dswing.aatext=true

> Configure project :
Current version: 21.2.1 Game: blokus_2021 (Java version: 11)
Java version 1.8 is recommended - expect issues with generating documentation (consider using '-x doc' if you don't care)

> Task :server:compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /home/sven/development/sc/backend/server/src/sc/server/gaming/GameRoom.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 4s
10 actionable tasks: 10 executed
```
in this branch
```
$ ./gradlew clean  && ./gradlew :test-client:jar
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=on -Dswing.aatext=true

> Configure project :
Current version: 21.2.1 Game: blokus_2021 (Java version: 11)
Java version 1.8 is recommended - expect issues with generating documentation (consider using '-x doc' if you don't care)

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 721ms
6 actionable tasks: 5 executed, 1 up-to-date
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=on -Dswing.aatext=true

> Configure project :
Current version: 21.2.1 Game: blokus_2021 (Java version: 11)
Java version 1.8 is recommended - expect issues with generating documentation (consider using '-x doc' if you don't care)

> Task :server:compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /home/sven/development/sc/backend/server/src/sc/server/gaming/GameRoom.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 2s
9 actionable tasks: 9 executed
```

File: helpers/test-client/build.gradle.kts (hunk 1 above)
Comment (xerus2000) [reply helpers/test-client/build.gradle.kts:17]: when we have time ^^

**Comment by xerus2000** [comment]
That is interesting, I get this on master:
```sh
❯ ./gradlew :test-client:jar
Picked up _JAVA_OPTIONS: -Djava.util.prefs.userRoot=/home/janek/.config/java
Configuration on demand is an incubating feature.

> Configure project :
Current version: 21.2.1 Game: blokus_2021 (Java version: 1.8)

FAILURE: Build failed with an exception.

* Where:
Build file '/home/janek/data/1-projects/software-challenge/gui/backend/server/build.gradle.kts' line: 3

* What went wrong:
An exception occurred applying plugin request [id: 'org.gradle.application']
> Failed to apply plugin class 'org.gradle.api.distribution.plugins.DistributionPlugin'.
   > Project#afterEvaluate(Action) on project ':server' cannot be executed in the current context.
```

**Comment by xerus2000** [comment]
Java version isn't the issue either:
```sh
❯ ./gradlew :test-client:jar
Picked up _JAVA_OPTIONS: -Djava.util.prefs.userRoot=/home/janek/.config/java
Starting a Gradle Daemon, 1 incompatible Daemon could not be reused, use --status for details
Configuration on demand is an incubating feature.

> Configure project :
Current version: 21.2.1 Game: blokus_2021 (Java version: 11)
Java version 1.8 is recommended - expect issues with generating documentation (consider using '-x doc' if you don't care)

FAILURE: Build failed with an exception.

* Where:
Build file '/home/janek/data/1-projects/software-challenge/gui/backend/server/build.gradle.kts' line: 3

* What went wrong:
An exception occurred applying plugin request [id: 'org.gradle.application']
> Failed to apply plugin class 'org.gradle.api.distribution.plugins.DistributionPlugin'.
   > Project#afterEvaluate(Action) on project ':server' cannot be executed in the current context.
```

**Comment by xerus2000** [comment]
Interesting, combining it with clean makes it work:
```sh
❯ ./gradlew clean :test-client:jar
Picked up _JAVA_OPTIONS: -Djava.util.prefs.userRoot=/home/janek/.config/java
Configuration on demand is an incubating feature.

> Configure project :
Current version: 21.2.1 Game: blokus_2021 (Java version: 1.8)

> Task :server:compileJava
Note: /home/janek/data/1-projects/software-challenge/gui/backend/server/src/sc/server/network/PerspectiveAwareConverter.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /home/janek/data/1-projects/software-challenge/gui/backend/server/src/sc/server/gaming/GameRoom.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 9s
16 actionable tasks: 15 executed, 1 up-to-date
```
Either way, it was unreliable and is now better :)

**Pull request merged**